CMakeLists.txt

```diff
@@ -736,6 +738,37 @@ if(BUILD_FOR_HEXAGON)
   endif()
 endif()
 
+
+if ((CMAKE_CXX_COMPILER_ID STREQUAL "Clang") OR (CMAKE_CXX_COMPILER_ID STREQUAL "GNU"))
+  if(USE_LLD)
+    find_program(LLD_PATH "lld")
+    if(LLD_PATH)
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fuse-ld=${LLD_PATH}")
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fuse-ld=${LLD_PATH}")
+    else()
+      if(USE_LLD STREQUAL "AUTO")
+        message(STATUS "Not using lld linker, executable was not found")
+      elseif(USE_LLD MATCHES ${IS_TRUE_PATTERN})
+        message(FATAL_ERROR "Could not find 'lld' executable but USE_LLD was set to ON")
+      endif()
+    endif()
+  endif()
+
+  if(USE_MOLD)
+    find_program(MOLD_PATH "mold")
+    if(MOLD_PATH)
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fuse-ld=${MOLD_PATH}")
```

Comment:
I would suggest using `CMAKE_SHARED_LINKER_FLAGS` (and other variants) rather than `CMAKE_CXX_FLAGS`. The former will only be applied when linking the library, while the latter will be apply to all compiler invocations (you will probably get an unused flag warning).